packages/stellar-axelar-std/src/traits.rs

```diff
@@ -1,3 +1,28 @@
+extern crate alloc;
+
+use alloc::vec::Vec;
+
+pub trait StringExt {
+    fn is_ascii(&self) -> bool;
+}
+
+impl StringExt for soroban_sdk::String {
+    fn is_ascii(&self) -> bool {
+        let mut bytes: Vec<u8> = Vec::new();
+        bytes.resize(self.len() as usize, 0);
+        self.copy_into_slice(&mut bytes);
+
+        for &byte in bytes.iter() {
+            let character = byte as char;
```

Comment:
Casting to char doesn't seem safe/well defined, since it expects a valid unicode value
https://doc.rust-lang.org/std/primitive.char.html#validity-and-layout

Check the byte value directly instead?